Add pagination/count support to docs (close #277) (#2874)
hasura/graphql-engine: docs/graphql/manual/queries/pagination.rst

        ]
      }
    }

> Fetch aggregate count with limited results

Reviewer: ```suggestion
Fetch aggregate count with limited results in the same query
```

------------------------------------------

> ``count`` (as part of aggregations) can be used to do pagination.

Reviewer: ```suggestion
``count`` (*as part of aggregations*) can be used along with `limit` based pagination in the same query. This information can be used to calculate the total number of pages.
```

**Example:** Fetch a list of articles where a certain condition is true and get their count. Then limit the number of articles to display.

.. graphiql::
  :view_only:
  :query:
    query articles ($where: articles_bool_exp!) {
      articles_aggregate(where: $where) {
        aggregate {
          totalCount: count
        }
      }

>       articles (where: $where limit: 10 order_by: {rating: desc}) {

Reviewer: ```suggestion
      articles (where: $where limit: 4 order_by: {rating: desc}) {
```

Reviewer: This because you're only getting 4 results in the response, which doesn't add up to the total and limit values.
